bundles/org.openhab.binding.qbus/src/main/java/org/openhab/binding/qbus/internal/QbusBridgeHandler.java

```
@@ -0,0 +1,241 @@
+        Configuration config = this.getConfig();
+        scheduler.submit(() -> {
+            qbusComm = new QbusCommunication();
+
+            // Set callback from Qbus object to this bridge to be able to take bridge
+            // offline when non-resolvable communication error occurs.
+            setBridgeCallBack();
+
+            qbusComm.startCommunication();
+            if (!qbusComm.communicationActive()) {
+                qbusComm = null;
+                bridgeOffline();
+                return;
+            }
+
+            updateStatus(ThingStatus.ONLINE);
+
+            Integer refreshInterval = ((Number) config.get(CONFIG_REFRESH)).intValue();
+            setupRefreshTimer(refreshInterval);
+
+        });
+    }
+
+    private void setBridgeCallBack() {
+        this.qbusComm.setBridgeCallBack(this);
+    }
+
+    /**
+     * Schedule future communication refresh.
+     *
+     * @param interval_config Time before refresh in minutes.
+     */
+    private void setupRefreshTimer(Integer refreshInterval) {
+        if (this.refreshTimer != null) {
+            this.refreshTimer.cancel(true);
+            this.refreshTimer = null;
+        }
+
+        if ((refreshInterval == null) || (refreshInterval == 0)) {
+            return;
+        }
+
+        // This timer will restart the bridge connection periodically
+        logger.debug("Qbus: Checking for Client communication every {} min", refreshInterval);
+        this.refreshTimer = scheduler.scheduleWithFixedDelay(() -> {
+            logger.debug("Qbus: check communication after timerinterval");
+
+            if (!qbusComm.communicationActive()) {
+                logger.debug("Qbus: Restarting communication");
+                qbusComm.restartCommunication();
+
+                if (!qbusComm.communicationActive()) {
+                    qbusComm = null;
+                    bridgeOffline();
+                    updateStatus(ThingStatus.OFFLINE);
```

Can you set the ThingStatusDetail and a message why this is offline? You could retrieve a message by letting `communicationActive()` throw an exception, instead of returning a boolean.